vtex/faststore
# fix: fuzzy and operator usage when handling IS requests

## What's the purpose of this pull request?

This PR aims to correct fuzzy and operator usage when handling Intelligent Search API requests.

Previously, we specified a default value for fuzzy `fuzzy=auto`. But we shouldn’t include it on the first request to Search. According to the IS team, the correct behavior should let the search engine decide the best parameters for the search.

Re-applies changes made from https://github.com/vtex/faststore/pull/2204 without adding the blink effect added previously.


- [x] Adds fuzzy & operator parameters in the query
- [x] Avoid re-rendering the page
- [x] Check search PLP (blink issue)
- [x] Check category PLP (blink issue)
- [x] Check duplicate `search.query` event to IS

## How it works?

More details on https://github.com/vtex/faststore/pull/2204 description.

## How to test it?

Run the project locally or test though this [preview link](https://sfj-8636169--starter.preview.vtex.app)

### Testing search PLP

1. Search for `apple` using the input search field
2. Open the `Network Tab` and filter for `ClientProductGalleryQuery`
- In the first request ->`Response` tab: look for `metadata` object, the `fuzzy` and `logicalOperator` values
<img width="1489" alt="image" src="https://github.com/user-attachments/assets/3bc1ccf5-65f9-454a-bddf-8fac768007a9">

- Check if the same values is being set in `selectedFacets`
<img width="1227" alt="image" src="https://github.com/user-attachments/assets/383fcf0b-a616-472f-8266-478e145ebaf0">

- You shouldn't see any blink effect in the page in this scenario:

https://github.com/user-attachments/assets/c92aec29-d089-45d0-a1d0-8bc991c97bb1

### Testing in Category PLP
1. Click on `Technology` category
2. You shouldn't see any blink effect in the page in this scenario

### Duplicate Search event in PLP
1. Search for `apple` using the input search field (at once)
2. Open the `Network Tab` and filter for `event`
3. You shouldn't be able to see the event (with the same url and term) being send twice
<img width="1574" alt="image" src="https://github.com/user-attachments/assets/7f0607ac-02cf-4c1d-b201-1cddc5000677">


### Starters Deploy Preview

https://github.com/vtex-sites/starter.store/pull/617

## References

https://github.com/vtex/faststore/pull/2204
